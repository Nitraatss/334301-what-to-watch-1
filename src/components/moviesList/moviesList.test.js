// Core
import React from "react";
import renderer from "react-test-renderer";

// Component
import MoviesList from "./moviesList.jsx";

const mocks = {
  films: [
    {
      id: `1`,
      title: `John Wick`,
      genre: [`Crime`],
      preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
    },
    {
      id: `2`,
      title: `Killers Bodyguard`,
      genre: [`Crime`, `Comedy`],
      preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
    },
    {
      id: `3`,
      title: `Star Wars`,
      genre: [`Sci-Fi`],
      preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
    },
    {
      id: `4`,
      title: `The Grand Budapest Hotel`,
      genre: [`Drama`, `Comedy`],
      preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
    },
    {
      id: `5`,
      title: `The Cabin in the Woods`,
      genre: [`Thriller`],
      preview: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
    }
  ]
};

describe(`MoviesList:`, () => {
  it(`Correctly renders after relaunch`, () => {
    const tree = renderer.create(<MoviesList films={mocks.films} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
