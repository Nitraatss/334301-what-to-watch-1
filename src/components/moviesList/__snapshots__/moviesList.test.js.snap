// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList: Correctly renders after relaunch 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[Function]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="John Wick"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        John Wick
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[Function]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Killers Bodyguard"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Killers Bodyguard
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[Function]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Star Wars"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        Star Wars
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[Function]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="The Grand Budapest Hotel"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        The Grand Budapest Hotel
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
  >
    <button
      className="small-movie-card__play-btn"
      onClick={[Function]}
      type="button"
    >
      Play
    </button>
    <div
      className="small-movie-card__image"
    >
      <img
        alt="The Cabin in the Woods"
        height="175"
        src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
      >
        The Cabin in the Woods
      </a>
    </h3>
  </article>
</div>
`;
